gst/nnstreamer/tensor_filter/tensor_filter_common.c

```diff
@@ -2942,33 +2958,34 @@ nnstreamer_filter_shared_model_remove (void *instance, const char *key,
     void (*free_callback) (void *))
 {
   GstTensorFilterSharedModelRepresenatation *model_rep;
-  int ret = TRUE;
+  int ret = FALSE;
 
   /* search the table with key */
+  G_LOCK (shared_model_table);
   if (!shared_model_table) {
     ml_loge ("The shared model representation is not supported properly!");
-    return FALSE;
+    goto done;
   }
 
-  g_mutex_lock (&mutex);
   model_rep = g_hash_table_lookup (shared_model_table, key);
   if (!model_rep) {
     ml_loge ("There is no value of the key: %s", key);
-    g_mutex_unlock (&mutex);
-    return FALSE;
+    goto done;
   }
 
   /* remove instance from the list */
   model_rep->referred_list = g_list_remove (model_rep->referred_list, instance);
   ml_logd ("The referred instance of sharing key: %s has been removed!", key);
+  ret = TRUE;
 
   /* remove key from table if list is empty */
   if (g_list_length (model_rep->referred_list) == 0) {
     if (free_callback)
       free_callback (model_rep->shared_interpreter);
-    ret = g_hash_table_remove (shared_model_table, key);
+    g_hash_table_remove (shared_model_table, key);
```

Comment:
minor though, how about checking if either comment or the implementation should be changed?
`@return TRUE if the new key and value are removed. FALSE if failed to remove it.`

Reply:
updated